### PR TITLE
Fixing Nimfa/indian-pines/losses issues with the benchmark

### PR DESCRIPTION
**Pull request by cohenjer**
This PR fixes the following issues:
#10 : we make use of [new functionalities in benchopt](https://github.com/benchopt/benchopt/pull/461) to make the solvers decide which loss is used to track convergence. From the perspective of the objective, all losses (frobenius, kl, factor match score) are computed regardless of the solver.
#7  #9 : Indian pines is now provided in tensorly, so we use tensorly for the import. Much simpler code.
#6 : tensorly 8.0 is now released, so the factor match score computation is using optimal permutations of the factors thanks to scipy linear assignment solver.
#5 : Nimfa was running fully when iter_max=0 was provided. This is now fixed. Scikit-learn also does not run when 0 iterations are requested.

### CODE REVIEW

File: solvers/tensorly.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
 from benchopt import BaseSolver, safe_import_context
+from benchopt.stopping_criterion import SufficientProgressCriterion
 
 
 with safe_import_context() as import_ctx:
     import copy
     from tensorly.decomposition._nn_cp import non_negative_parafac
     from tensorly.decomposition._nn_cp import non_negative_parafac_hals
 
+#TODO fix??
```
Comment (cassiofragadantas) [inline solvers/tensorly.py:10]: Fix what?

File: solvers/nimfa.py [hunk 2]
```diff
@@ -8,20 +9,22 @@
...
     # any parameter defined here is accessible as a class attribute
     parameters = {
         'strategy': ['MU', 'ALS-PG'],
-        'loss': ['euclidean']  # Other choices: 'divergence' (for KL)
+        'loss': ['euclidean'],  # Other choices: 'divergence' (for KL)
+        'sub_iter_max': [10],
     }
 
-    stopping_strategy = "iteration"
+    #stopping_strategy = "iteration"
```
Comment (cassiofragadantas) [inline solvers/nimfa.py:22]: Delete this?

File: solvers/apg.py [hunk 3]
```diff
@@ -14,8 +15,9 @@ class Solver(BaseSolver):
     parameters = {
         'n_inner_iter': [1, 5]
```
Comment (cassiofragadantas) [inline solvers/apg.py:16]: Why a 'loss' parameter was included to the other solvers but not to this one?

**Review by cassiofragadantas — COMMENTED** [review]
Apart from the minor comments above, everything looks good to me!

**Comment by cohenjer** [comment]
Thanks @cassiofragadantas, I will handle this asap.

File: solvers/apg.py (hunk 3 above)
Comment (cohenjer) [reply solvers/apg.py:16]: I added it here as well

**Pull request merged**